### PR TITLE
Port CommunityToolkit.HighPerformance docs

### PR DESCRIPTION
**Pull request by Sergio0694**
This PR ports the Diagnostics docs. Haven't updated the links to the generated docs yet, same as #146.

### CODE REVIEW

**Comment by Sergio0694** [comment]
Hey @michael-hawker, could you take a look at this one and #146 when you have a minute? I'm getting close to publishing the 8.1.0 release and we should really have all docs being moved to the new place now, even if the redirects aren't there yet 😄
Thank you!

File: docs/high-performance/Introduction.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, get started, visual studio, high performance, net core, net standard
+---
+
+# Introduction to the High Performance package
+
+The `CommunityToolkit.HighPerformance` contains helpers and extensions to work in high-performance scenarios. This package can be installed through NuGet, and it has the following multi-targets:
+
+- .NET Standard 2.0
+- .NET Standard 2.1
+- .NET Core 3.1
```
Comment (michael-hawker) [inline docs/high-performance/Introduction.md:14]: .NET Core 3.1 is no longer supported eh? Didn't notice if this list was in the Diagnostics docs as well?

File: docs/high-performance/Introduction.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- .NET 6
+
+This means that you can use it from anything from UWP or legacy .NET Framework applications, games written in Unity, cross-platform mobile applications using Xamarin, to .NET Standard libraries and modern .NET Core 2.1 or .NET Core 3.1 applications. The API surface is almost identical in all cases, and lots of work has been put into backporting as many features as possible to older targets like .NET Standard 1.4 and .NET Standard 2.0. Except for some minor differences, you can expect the same APIs to be available on all target frameworks.
+
+The reason why multi-targeting has been used is to allow the package to leverage all the latest APIs on modern runtimes (like .NET 5) whenever possible, while still offering most of its functionalities to all target platforms. It also makes it possible for .NET Core 2.1 to use all the APIs that it has in common with .NET Standard 2.1, even though the runtime itself doesn't fully implement .NET Standard 2.1. All of this would not have been possible without multi-targeting to both .NET Standard as well as individual runtimes.
```
Comment (michael-hawker) [inline docs/high-performance/Introduction.md:19]: Might want to review this stuff too, should probably change .NET 5 to 6?

**Review by michael-hawker — CHANGES_REQUESTED** [review]
Should update the introduction for the new target frameworks.

File: docs/high-performance/Introduction.md (hunk 1 above)
Comment (Sergio0694) [reply docs/high-performance/Introduction.md]: Good catch! Fixed 🙂

**Review by michael-hawker — APPROVED** [review]

**Pull request merged**